### PR TITLE
Core/ts version task 423 get user order orders

### PR DESCRIPTION
**Pull request by franken5tein**
## What does this MR do?

Add end-to-end tests for get user order and user orders endpoints

## Check list  

- [x] All Merge Request pipelines should be passed.
- [x] Make sure all the changes are necessary and correct.
- [x] Make sure no needed change missing (all necessary changes are made).
- [x] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [ ] All necessary unit tests are written.
- [x] Critical changes should be tested locally (or on staging).

### CODE REVIEW

File: ts-version/src/test/e2e/get-user-orders.e2e.test.ts [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    await providers.stopProviders();
+    spiedSetInterval.mockRestore();
+  });
+
+  it("Should return correct error when not logged in", async () => {
+    const getUserOrderInput = {
+      market: "WETH-DAI",
+      id: 1,
+      // start_time: new Date(),
+      // end_time: new Date(),
```
Comment (GreatGaryCooper) [inline ts-version/src/test/e2e/get-user-orders.e2e.test.ts:39]: Please remove these comments.

File: ts-version/src/test/e2e/get-user-orders.e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        headers: { Authorization: `Bearer ${jwtToken}` },
+      },
+    );
+    // end create new order
+
+    const getUserOrderInput = {
+      market: orderResponse.data._market,
+      id: orderResponse.data.id,
+      //   start_time: orderResponse.data.created_at,
+      //   end_time: new Date(),
```
Comment (GreatGaryCooper) [inline ts-version/src/test/e2e/get-user-orders.e2e.test.ts:121]: Also these.

File: ts-version/src/test/e2e/get-user-orders.e2e.test.ts [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    expect(response.data[0].id).toEqual(1);
+    expect(response.data[0].market).toEqual(createOrderPayload.market);
+    expect(response.data[0].side).toEqual(createOrderPayload.side);
+    expect(math.floor(response.data[0].price).toString()).toEqual(
+      createOrderPayload.price,
+    );
+    expect(new Decimal(response.data[0].base_quantity)).toEqual(
+      math.div(createOrderPayload.amount, math.pow(10, 18)),
+    );
+    expect(Number(response.data[0].quote_quantity).toFixed()).toEqual(
```
Comment (GreatGaryCooper) [inline ts-version/src/test/e2e/get-user-orders.e2e.test.ts:146]: Can you add a short comment here, explaining what this case is about?
Or maybe create a helper function to do the calculations and then check the result...

File: ts-version/src/test/e2e/get-user-order.e2e.test.ts [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+    const response = await axios.get(
+      `http://localhost:${serverPort}/user/order`,
+      {
+        params: getUserOrderInput,
+        headers: { Authorization: `Bearer ${jwtToken}` },
+      },
+    );
+
+    console.log("response.data", response.data);
```
[the remark this answers is not in the document]
Comment (franken5tein) [reply ts-version/src/test/e2e/get-user-order.e2e.test.ts:127]: Done!

File: ts-version/src/test/e2e/get-user-order.e2e.test.ts
[the remark this answers is not in the document]
Comment (franken5tein) [reply ts-version/src/test/e2e/get-user-order.e2e.test.ts:1]: ✌️👽️

File: ts-version/src/test/e2e/get-user-orders.e2e.test.ts [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    const createdOrder = createOrderRes.data.data;
+    const response = await axios.get(
+      `http://localhost:${serverPort}/user/orders`,
+      {
+        headers: { Authorization: `Bearer ${jwtToken}` },
+        params: {
+          market,
+          id: createdOrder.id,
+          //   start_time: createdOrder.created_at,
+          //   end_time: new Date(),
```
Comment (GreatGaryCooper) [inline ts-version/src/test/e2e/get-user-orders.e2e.test.ts:109]: Please remove this comments

File: ts-version/src/test/e2e/get-user-orders.e2e.test.ts [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+    const createOrderRes = await createOrder();
+    const createdOrder = createOrderRes.data.data;
+
+    const response = await axiosInstance.get("/user/orders", {
+      headers: { Authorization: `Bearer ${jwtToken}` },
+      params: {
+        market: createdOrder.market,
+        id: createdOrder.id + 1,
+        // start_time: createdOrder.start_time,
+        // end_time: new Date()
```
Comment (GreatGaryCooper) [inline ts-version/src/test/e2e/get-user-orders.e2e.test.ts:145]: Also these

**Review by GreatGaryCooper — APPROVED** [review]

**Review by WilliamHawkingStephen — APPROVED** [review]

**Review by GreatGaryCooper — APPROVED** [review]
Awesome test cases

**Pull request merged**